helm/idol-answerserver/templates/_container.tpl

```diff
@@ -0,0 +1,30 @@
+# BEGIN COPYRIGHT NOTICE
+# Copyright 2024 Open Text.
+# 
+# The only warranties for products and services of Open Text and its affiliates and licensors
+# ("Open Text") are as may be set forth in the express warranty statements accompanying such
+# products and services. Nothing herein should be construed as constituting an additional warranty.
+# Open Text shall not be liable for technical or editorial errors or omissions contained herein.
+# The information contained herein is subject to change without notice.
+#
+# END COPYRIGHT NOTICE
+
+{{- define "idolanswerserver.container.base" -}}
+{{/* specific merges to container specification */}}
+{{- $root := get . "root" | required "idolanswerserver.container.base: missing root" -}}
+{{- $component := get . "component" | required "idolanswerserver.container.base: missing component" -}}
+{{- /* END of idolanswerserver.container.base */ -}}
+{{- end -}}
+
+{{- define "idolanswerserver.container" -}}
+{{- $root := get . "root" | required "idolanswerserver.container: missing root" -}}
+{{- $component := get . "component" | required "idolanswerserver.container: missing component" -}}
+{{- include "idol-library.util.merge" (dict
+  "root" $root
+  "component" $component
+  "source" "idol-library.aciserver.container.base.v1"
+  "destination" "idolanswerserver.container.base" 
+  "env" (list (dict "name" "IDOL_COMMAND_PARAMS" "value" "-idolcluster"))
+  "volumeMounts" (list (dict "name" "licensing" "mountPath" "/answerserver/license"))
```

Comment:
Will need to persist both the license and uid folders.
Wonder if it's worth making the volume available to use for caching model files too?
Should be possible to use one volume claim and then mount multiple subPaths from within that (see how it's done in the `idol-nifi` chart) - probably want to rename the volumeClaimTemplates to `state` or similar

Reply:
https://github.com/opentext-idol/idol-containers-toolkit/pull/63/commits/c4d6d663accc17e2d6b5659d9f067f13c5d5574a For the model cache I've used the [default hugging face cache directory](https://teams.microsoft.com/l/message/19:e1d6a258680c4c73a25cdd5cfb9097bd@thread.tacv2/1721293683018?tenantId=10a18477-d533-4ecd-a78d-916dbd849d7c&groupId=bc988af4-1562-4554-bb99-ec7ac30b1738&parentMessageId=1721293683018&teamName=IDOL%20Text&channelName=F456017%20Search%20Abstractor%20conversations&createdTime=1721293683018), replacing `~` with `/home/idoluser` because `~` doesn't look like it works. Did you want this to be configurable via `values.yaml`?